Add missing App Permission Types

fabianthdev opened the pull request:
The original forked project only supports three permission types: Photos, Background Audio and Background Refresh.

However, according to the [AltSource specification](https://noah978.gitbook.io/altstore-docs/sources#releasing-apps) there are 13 additional valid permission types. This PR adds a localized name and icon for:
- Camera
- Location
- Contacts
- Reminders
- Apple Music
- Microphone
- Speech Recognition
- Bluetooth
- Network
- Calendars
- TouchID
- FaceID
- Siri
- Motion

Also, all icons are now using SF Symbols instead of the old icons from the AltStore developers.

> :warning: **Please Note**: To be able to use the most fitting system icons for the permission types, I've bumped the target OS version for the AltStoreCore target from iOS 12.2 to iOS 14.0. This should not interfere with SideStore as its target OS version has also been increased to iOS 14 in a recent commit.

fabianthdev commented:
I am not sure about the symbols for the Siri and Bluetooth permission types. SF Symbols does not include official icons for these two types, but I could create some and add them to this PR.
I would like to hear your opinion on this and the other SF Symbols I've chosen for the permission types.

JJTech0130 commented:
Oops dumb mobile

JJTech0130 commented:
Was going to say does Swift Playgrounds have the  icons? It has permissions picker iirc

JJTech0130 commented:
Also, it would be nice to eventually pull out *all* entitlements present in the IPA, in addition to the ones explicitly declared. That would mean we'd have to handle some without icons

jkcoxson commented on AltStoreCore/Model/AppPermission.swift:
```diff
@@ -25,18 +25,44 @@ public extension ALTAppPermissionType
+        case .calendars: return NSLocalizedString("Calendars", comment: "")
+        case .touchID: return NSLocalizedString("Touch ID", comment: "")
+        case .faceID: return NSLocalizedString("Face ID", comment: "")
+        case .siri: return NSLocalizedString("Siri", comment: "")
```
The Xcode signer won't sign apps with Siri entitlements. Are you sure that SideStore can resign apps with this?

JoeMatt replied:
Doesn't hurt to have them in there for now in case that ever changes. 

The code should strip that entitlement in the prepare stage if not already. 

Not even sure if that's possible, but still, having the key/value mappings here is harmless regardless I'm assuming?

JoeMatt replied:
We could also extend the enum with a 'var isSupported: Bool { switch self...}'

fabianthdev commented:
> Was going to say does Swift Playgrounds have the icons? It has permissions picker iirc

I just had a quick look around for Swift frameworks that use SF symbols to describe iOS permissions, and found this:
https://github.com/jevonmao/PermissionsSwiftUI/blob/main/Sources/CorePermissionsSwiftUI/Store/ComponentsStore/PermissionComponentsStore.swift

They use very similar icons to the ones I've selected, and even almost the same one for the bluetooth permission. Only the icons for speech recognition and Siri are very different from the ones I've chosen:
- Speech recognition: waveform.and.mic (me) vs. rectangle.3.offgrid.bubble.left.fill (they)
- Siri: mic.and.signal.meter (me) vs. waveform (they)

I could change the icons to the ones they are using in the linked framework.

fabianthdev commented:
I've just added two small tweaks: I've added the missing icon and text for the camera permission and switched the icons to their filled version if available for a more cohesive look.
